File: Models/Patient.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace sem5pi_24_25_g051.Models
{
    public class Patient
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public required string FirstName { get; set; }
        [Required]
        public required string LastName { get; set; }
        [Required]
        public string FullName => $"{FirstName} {LastName}";
        [Required]
        public DateTime DateOfBirth { get; set; }
        [Required]
        public required string Gender { get; set; }
        [Required]
        public required string MedicalRecordNumber { get; set; }
        [Required]
        public required string Email { get; set; }
        [Required]
        [Phone]
        public required string Phone { get; set; }

        [Required]
        public required string AllergiesOrMedicalConditions { get; set; }

        [Required]
        [Phone]
        public required string EmergencyContact { get; set; }
       /*
        public void RequestDataAccess() { }
        public void RequestDataRemoval() { }     */
    }
}

File: patientManagement/src/Models/Patient.js
const mongoose = require('mongoose');

const appointmentSchema = new mongoose.Schema({
    date: { type: Date, required: true },
    type: { type: String, required: true },
    status: { type: String, enum: ['Scheduled', 'Completed', 'Cancelled'], required: true },
});

const patientSchema = new mongoose.Schema({
    firstName: String,
    lastName: String,
    fullName: { type: String, get: function() { return `${this.firstName} ${this.lastName}`; } },
    dateOfBirth: Date,
    gender: String,
    medicalRecordNumber: { type: String, unique: true },
    contactInfo: {
        email: String,
        phone: String,
    },
    allergies: [{ type: mongoose.Schema.Types.ObjectId, ref: 'Allergy', required: true }],
    medicalConditions: [String],
    emergencyContact: {
        name: String,
        phone: String,
    },
    appointmentHistory: [appointmentSchema],
});

module.exports = mongoose.model('Patient', patientSchema);
